### PR TITLE
Remake cluster stress tests

### PR DESCRIPTION
**Pull request by martinfurmanski**

### CODE REVIEW

File: stresstests/src/test/java/org/neo4j/causalclustering/stresstests/BackupLoad.java [hunk 1]
```diff
@@ -30,43 +28,37 @@
...
-import org.neo4j.causalclustering.discovery.Cluster;
 
-class BackupLoad extends RepeatUntilOnSelectedMemberCallable
+class BackupLoad extends RepeatOnRandomMember
 {
-    private final Predicate<Throwable> isTransientError =
-            new IsConnectionException().or( new IsConnectionRestByPeer() ).or( new IsChannelClosedException() )
-                    .or( new IsStoreClosed() );
+    private final Predicate<Throwable> isTransientError = new IsConnectionException()
+            .or( new IsConnectionRestByPeer() )
```
Comment (hugofirth) [inline stresstests/src/test/java/org/neo4j/causalclustering/stresstests/BackupLoad.java]: I know this wasn't caused by this PR, but could maybe roll in a refactor for this typo?

File: stresstests/src/test/java/org/neo4j/causalclustering/stresstests/Resources.java [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+import static org.neo4j.causalclustering.stresstests.ClusterConfiguration.configureRaftLogRotationAndPruning;
+import static org.neo4j.causalclustering.stresstests.ClusterConfiguration.enableRaftMessageLogging;
+import static org.neo4j.causalclustering.stresstests.Config.NUMBER_OF_CORES;
+import static org.neo4j.causalclustering.stresstests.Config.NUMBER_OF_EDGES;
+import static org.neo4j.causalclustering.stresstests.Config.TX_PRUNE;
+import static org.neo4j.causalclustering.stresstests.Config.WORKING_DIR;
+import static org.neo4j.helper.DatabaseConfiguration.configureTxLogRotationAndPruning;
+import static org.neo4j.helper.StressTestingHelper.ensureExistsAndEmpty;
+
+class Resources
```
Comment (hugofirth) [inline stresstests/src/test/java/org/neo4j/causalclustering/stresstests/Resources.java]: This seems fairly similar to `ClusterRule.java`. Would it be worth Rewriting this as a factory method for producing a cluster rule with Stresstest config, or even factoring out the common elements of both (if `Resources` technically being a `TestRule` is bad) to reduce the maintenance footprint? Happy to pick this up in a follow up PR, just wanted to put this here to start the conversation.

File: stresstests/src/test/java/org/neo4j/causalclustering/stresstests/StartStopLoad.java [hunk 3]
```diff
@@ -21,46 +21,45 @@
...
-        ClusterMember member = isCore ? cluster.getCoreMemberById( id ) : cluster.getReadReplicaById( id );
         File storeDir = member.database().getStoreDir();
-        KernelExtensions kernelExtensions =
-                member.database().getDependencyResolver().resolveDependency( KernelExtensions.class );
+        KernelExtensions kernelExtensions = member.database().getDependencyResolver().resolveDependency( KernelExtensions.class );
+        log.info( "Stopping: " + member );
         member.shutdown();
         assertStoreConsistent( storeDir, kernelExtensions );
-        LockSupport.parkNanos( 5_000_000_000L );
+        Thread.sleep( 5000 );
```
Comment (hugofirth) [inline stresstests/src/test/java/org/neo4j/causalclustering/stresstests/StartStopLoad.java:61]: Is this sleep strictly necessary? Could the consistency check be wrapped in a completable future or something? I suppose with workloads taking this long 5 seconds isn't that big of a deal though.

Comment (martinfurmanski) [reply stresstests/src/test/java/org/neo4j/causalclustering/stresstests/StartStopLoad.java:61]: The sleep has nothing to do with the consistency check really. It is just a rate-limiter for the speed at which we start and stop instances. I could imagine perhaps making the sleep-time randomized to hit various cases due to the shifts in timing in the stress testing. I think the consistency check might also be made randomized (e.g. happens on scale from 0 to 100% of the time).

File: stresstests/src/test/java/org/neo4j/causalclustering/stresstests/Resources.java (hunk 2 above)
Comment (martinfurmanski) [reply stresstests/src/test/java/org/neo4j/causalclustering/stresstests/Resources.java]: I think the idea makes sense, and would definitely appreciate if you put your touch on this later.

**Review by RagnarW — COMMENTED** [review]
Looks a lot better!

File: stresstests/src/test/java/org/neo4j/causalclustering/stresstests/StartStopLoad.java (hunk 3 above)
Comment (hugofirth) [reply stresstests/src/test/java/org/neo4j/causalclustering/stresstests/StartStopLoad.java:61]: Ah I see - my mistake.

**Pull request merged**